daemon/daemon_unix.go

```diff
@@ -653,6 +653,9 @@ func configureKernelSecuritySupport(config *Config, driverName string) error {
 }
 
 func (daemon *Daemon) initNetworkController(config *Config, activeSandboxes map[string]interface{}) (libnetwork.NetworkController, error) {
+	if err := processPoolsConfig(config); err != nil {
+		return nil, err
+	}
```

Comment:
One can configure `--default-address-pools` and decide to change it on a daemon reload. We can allow that change only if all the networks created before have been deleted. So the libnetwork controller has to be instantiated. But currently ipamutils is an independent package.

How about letting libnetwork controller get the `--default-address-pools` config, do the required validity checks and then call ipamutils.InitAddressPools with the new config or fail the controller creation if the validity checks fail ?